hydrabus_framework/core/engine.py

```diff
@@ -0,0 +1,105 @@
+import inspect
+import pkgutil
+import sys
+
+from importlib import import_module
+from prompt_toolkit.shortcuts import PromptSession
+from prompt_toolkit.styles import Style
+
+from hydrabus_framework.core.dispatcher import Dispatcher
+from hydrabus_framework.modules.amodule import AModule
+from hydrabus_framework.utils.logger import Logger
+
+
+__author__ = "Jordan Ovrè <ghecko78@gmail.com>"
+
+
+class HydraFramework:
+    """
+    Framework core engine
+    """
+    def __init__(self):
+        self.logger = Logger()
+        self.app_path = sys.path[0]
+        self.current_module = None
+        self.modules = self._list_modules()
+        self.modules_history = []
+        self.dispatcher = Dispatcher()
+        self.prompt_style = Style.from_dict({
+            # User input (default text).
+            '': '#ffffff',
+
+            # Prompt.
+            'base': '#3399ff',
+            'pound': '#3399ff',
+            'module': '#ff0000 bold',
+            'category': '#ffffff',
+        })
+        self.prompt = [
+            ('class:base', '[hbf] '),
+            ('class:module', ''),
+            ('class:category', ''),
+            ('class:pound', '> '),
+        ]
+
+    def update_prompt(self, module_name):
```

Comment:
Due to the fact that your commands take as parameter, would it be possible to store the current module as a properties of your class? So you would be able to update it from your command and call the update_command from your main loop? It will reduce the coupling

Reply:
Done [#6bfd685](https://github.com/hydrabus-framework/framework/commit/6bfd6851b4ea79dac0de35d717766e6dcebee25b)